File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.swift
import Foundation

func solution(_ numer1:Int, _ denom1:Int, _ numer2:Int, _ denom2:Int) -> [Int] {
    var numer = numer1*denom2 + numer2*denom1
    var denom = denom1*denom2
    
    var div = 1
    for n in 1...denom {
        if (denom%n == 0) && (numer%n == 0) { div = n }
    } 
    return [numer/div, denom/div]
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = {0,0};
        
        answer[1] = denom1*denom2;
        answer[0] = numer1*denom2 + numer2*denom1;
        
        int div = 1;
        for(int i=2; i<=1000000; i++) {
            if(answer[0]%i == 0 && answer[1]%i == 0) div = i;
        }
        
        answer[0] /= div;
        answer[1] /= div;
        return answer;
    }
}